### ISSUE TITLE
nan AUC

### ISSUE THREAD

**Issue by loloCFC** [lyst/lightfm#339]
Good day.  I am a student currently working on a recommender system and since I have no experience on such systems or machine learning I have decided to use your library LightFM since it is well documented and easy to set up. After trying it out I ran into a problem which gave me nans in my AUC. This was troubling me a lot since I was planning to optimize my hyperparameters with the AUC and I cannot do that with nan values. I tried to look up why the model produces nan values but to no avail. I have checked my inputs and know that they do not consist nans nulls or nones. I would greatly appreciate your help in this matter

```
<35x209 sparse matrix of type '<class 'numpy.float32'>'
with 2557 stored elements in COOrdinate format>
<35x209 sparse matrix of type '<class 'numpy.float32'>'
with 2557 stored elements in COOrdinate format>
<209x645 sparse matrix of type '<class 'numpy.float32'>'
with 36784 stored elements in Compressed Sparse Row format>
<35x50 sparse matrix of type '<class 'numpy.float32'>'
with 560 stored elements in Compressed Sparse Row format>

AUC: [nan nan nan nan]
```
The first 2 matrices are the interaction matrix and its corresponding weight and the next 2 are the item features and user features matrix

**Comment by maciejkula** [comment]
The most common reasons for model divergence are:

1. Learning rate which is too high.
2. Feature weights which are too large.

Can you check those things first?

**Comment by loloCFC** [comment]
I kept tweaking the learning rate and feature weights but also to no avail

**Comment by maciejkula** [comment]
What values have you tried? Have you tried without user/item features?

**Comment by loloCFC** [comment]
0.00001 for learning rate and 1e-6 for item features and 1e-2 for user features. These are not the only values I have tried though but most are relatively close to these values. Also removing either of the features still results in nan AUC. Funny thing is that when I only include only the item features the predictions are almost the same for all users but if I include both feature matrices then the predictions vary.

**Comment by maciejkula** [comment]
Is your test set nonempty?

**Comment by loloCFC** [comment]
do I need to have a test set? I mean I know I should but I tried to look at the stackexchange example and it was able to get an auc on the training set alone. I am trying to do the same thing but am getting nan values instead. This was the part before it had side information

**Comment by maciejkula** [comment]
Can you post the line of code you use to obtain the AUC values?

**Comment by loloCFC** [comment]
`food_auc = auc_score(model, data['interactions'], item_features=data['item_features'], user_features=data['user_features'], num_threads=2)`

interactions here is assumed to be my training set

**Comment by maciejkula** [comment]
I don't know the immediate reason why you are getting the result.

Your dataset is extremely small, much smaller than anything that can be reasonably used in a real application. For this reason you may be hitting an edge case that I have not seen before. If you figure out what the problem is, let me know and we can make sure what this is.

**Comment by SimonCW** [comment]
I’m closing this issue because it has been inactive for a long time. If you still encounter the problem, please open a new issue.

Thank you!
